src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java

```diff
@@ -568,7 +575,7 @@ private int getCandidateIndexForInitialConnection() {
         return candidateReaderIndex;
       }
     }
-    return this.hosts.get(WRITER_CONNECTION_INDEX) != null ? WRITER_CONNECTION_INDEX : NO_CONNECTION_INDEX;
+    return WRITER_CONNECTION_INDEX;
```

Comment:
Don't we need to return NO_CONNECTION_INDEX in case topology is empty?

Reply:
The only function that calls this one has a few steps that it takes before calling this function, and among those steps it verifies that the topology is not null or empty. So this function is not being called until after that has already been verified. However I'll add another check here in case we add more code that calls this function in the future